sklearn/metrics/tests/test_regression.py

```diff
@@ -611,3 +620,37 @@ def test_pinball_loss_relation_with_mae():
         mean_absolute_error(y_true, y_pred)
         == mean_pinball_loss(y_true, y_pred, alpha=0.5) * 2
     )
+
+
+def check_array_api_compute_metric(name, metric, array_namepsace, _device, dtype):
+    xp, _device, dtype = _array_api_for_tests(array_namepsace, _device, dtype)
+    y_true_np = np.array([[1, 3], [1, 2]], dtype=float)
+    y_pred_np = np.array([[1, 4], [1, 1]], dtype=float)
+    y_true_xp = xp.asarray(y_true_np, device=_device)
+    y_pred_xp = xp.asarray(y_pred_np, device=_device)
+    metric_np = metric(y_true_np, y_pred_np)
+
+    with config_context(array_api_dispatch=True):
+        metric_xp = metric(y_true_xp, y_pred_xp)
+        assert metric_xp.shape == ()
+        assert metric_xp.dtype == y_true_xp.dtype
+        target_device = "<CUDA Device 0>" if "cupy" in xp.__name__ else "cpu"
+        assert str(device(metric_xp)) == target_device  # R2_score gets moved to CPU
+
+        assert_allclose(
+            _convert_to_numpy(metric_xp, xp=xp),
+            metric_np,
+            atol=np.finfo(dtype).eps * 100,
+        )
+
+
+@pytest.mark.parametrize(
+    "array_namespace, _device, dtype", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "check",
+    [check_array_api_compute_metric],
+)
+def test_r2_score_array_api_compliance(check, array_namespace, _device, dtype):
+    name = r2_score.__class__.__name__
+    check(name, r2_score, array_namespace, _device=_device, dtype=dtype)
```

Comment:
Thanks for the updates.

However, instead of introducing a new function could you please consolidate this into the `sklearn/metrics/tests/test_common.py` framework similarly to what was done in:


https://github.com/scikit-learn/scikit-learn/pull/27137/files#diff-242ab1f5597b90a66238803ae803d8788d9a09c1fc3d698a13b6e7bb002941d0R1779-R1788

Reply:
BTW, since there are no check for regression metric yet, you will have to define your own by taking inspiration on the classification checkers introduced in #2731.

Reply:
Done :+1: